### PR TITLE
refuse requests when the server is too busy

### PR DESCRIPTION
**Pull request by lloyd**
issue #2722

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@lloyd - It'd be awesome to have 2 fewer bugs listed but this needs to be rebased and @jrgm is worried that 20ms is a too low of a default.

**Comment by lloyd** [comment]
absolutely, I'm on the hook to rebase and address @jrgm's concerns.  I'll do this today.

File: lib/load_gen/activities/add_email.js [hunk 1]
```diff
@@ -56,37 +56,27 @@ exports.startFunc = function(cfg, cb) {
...
-        return cb(msg);
-      }
+
       // now get the verification secret
       wcli.get(cfg, '/wsapi/fake_verification', context, {
         email: email
       }, function (err, r) {
+        err = common.checkResponse(err, r);
         if (err) return cb(err);
-        if (!r || r.code !== 200) {
```
Comment (jrgm) [inline lib/load_gen/activities/add_email.js:71]: Needs to check for err.code !== 'nonJSON' here.

File: lib/load_gen/activities/reset_pass.js [hunk 2]
```diff
@@ -65,31 +65,24 @@ exports.startFunc = function(cfg, cb) {
     pass: user.password,
     site: userdb.any(user.sites)
   }, function (err, r) {
+    err = common.checkResponse(err, r);
     if (err) return cb(err);
```
Comment (jrgm) [inline lib/load_gen/activities/reset_pass.js:69]: Needs to check for err.code !== 'nonJSON' here, I seem to recall.

**Comment by jrgm** [comment]
Hmm, github is having issues atm, and won't let me add another line note. But about line 72 is activities/signup, that also needs to check for nonJSON in response from fake_verification (or that's what I recall; sorry, thought I had added a comment here about these places to fix, but apparently did not).

**Comment by lloyd** [comment]
at this point I believe all open issues are addressed.  I did my best to load test this on AWS - https://gist.github.com/4489337

But load testing on AWS cannot quite excercise these failures.  I feel comfortable enough with this to merge and vet it in our staging environment, because the actual code changes are not that extensive.

I leave it up to @jrgm whether this is really merge worthy at this point.

**Pull request merged**